File: baekjoon/N1009.java
import java.util.Scanner;

public class N1009 {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int t = sc.nextInt();
		for (int i = 0; i < t; i++) {
			int ans = solve(sc.nextInt() % 10, sc.nextInt());
			if (ans == 0) {
				ans = 10;
			}
			System.out.println(ans);
		}
		sc.close();
	}

	static int solve(int a, int b) {
		if (b == 1) {
			return a;
		}
		int temp = solve(a, b / 2);

		if (b % 2 == 0) {
			return temp * temp % 10;
		}
		return temp * temp * a % 10;
	}
}

File: baekjoon/N1009.cpp
#include <stdio.h>
#include <cmath>
int solve(int a, int b)
{
	if (b == 1)
		return a % 10;
	int ret = solve(a, b / 2);
	ret *= ret;
	if (b % 2)
		ret *= a;
	return ret % 10;
}
int main()
{
	int t;
	scanf("%d", &t);
	while (t--)
	{
		int a, b;
		scanf("%d %d", &a, &b);
		int ans = solve(a, b);
		if (!ans)
			ans = 10;
		printf("%d\n", ans);
	}
}
